themarshallproject/klaxon
# Upgrade to Rails 7.0 and Ruby 3.1.4

This is a doozy — this PR drags Klaxon into 2023 by bringing it up to both Rails 7 and Ruby 3. A good bit of what's in this PR is housekeeping, but I'll attempt to make a semi-comprehensive list of the changes below.

### Pay special attention to

Here's what's been done:

- `.ruby-version` has been bumped up to `3.1.4`. Surprisingly very little conflicts dependency wise — helps that there are very few moving parts here.
- Got rid of `spring` and moved from `byebug` to `debug` just like we did in EndRun.
- A big part of this upgrade was trying to keep Heroku happy — so I made a number of changes that ensure we're still compatible there. The biggest change there was changing the default Heroku stack and moving over to its native version of "run this on release."
- `sass-rails` is gone in favor of `dartsass-rails`. This is one of the current blessed ways to interface with Sass in Rails these days. The other gem is long abandoned.
  -  As a result I also had to upgrade a ton of Sass files (most of the changes here) because Dart Sass doesn't like division using `/`. (We also have this warning all over EndRun.)
- Moved to `terser` from `uglifier`.
- Purged Turbolinks and didn't replace it with Turbo because we were never using it anyway.

### How to test

I think there are a couple things to test here — one I think you should do and one I think you should trust I did. 😅 

See if you can get it running locally still. I started with a clean DB and went through the local setup steps.

You could _also_ consider making sure it still can run on Heroku. I did do this using a personal account I have that had some credits still on it. Emails still send properly (not with SendGrid tho — they banned my account immediately ¯\\_(ツ)_/¯) I set up the scheduler and everything and it appears to work as expected.

I have a follow up to this PR that will take a swing at simplifying Docker stuff. I'd also like to update the README to push folks toward using Gmail as the email service instead.
